openedx/core/lib/xblock_utils.py

```diff
@@ -302,7 +302,9 @@ def add_staff_markup(user, has_instructor_access, disable_staff_debug_info, bloc
         # check that the course is a mongo backed Studio course before doing work
         is_studio_course = block.course_edit_method == "Studio"
 
-        if is_studio_course:
+        is_ccx_course = hasattr(course_id, 'ccx')
+        # preventing users from opening ccx courses in Studio, since Studio doesn't support CCX courses
+        if is_studio_course and not is_ccx_course:
             # build edit link to unit in CMS. Can't use reverse here as lms doesn't load cms's urls.py
             edit_link = "//" + settings.CMS_BASE + '/container/' + unicode(block.location)
```

Comment:
Instead of adding this toggle based on knowledge of the key, can we have CCX override `block. course_edit_method` on the blocks that it passes back so that it has `"CCX"` instead of `"Studio"`? We already have hooks for not displaying these fields for the purposes of old-style XML Modulestore courses -- might as well repurpose that logic, since it's less likely to break.